### PR TITLE
Fixing issue with logging in with private interface and the server never being ready to log in

### PR DESCRIPTION
**Pull request by rmoreira**
This fixes logging in to the server without using public ip address

Fixes https://github.com/test-kitchen/kitchen-ec2/issues/86

### CODE REVIEW

File: lib/kitchen/driver/ec2.rb [hunk 1]
```diff
@@ -89,20 +87,20 @@ def create(state)
...
-
         state[:server_id] = server.id
         info("EC2 instance <#{state[:server_id]}> created.")
         server.wait_for do
-          print '.'
-          # Euca instances often report ready before they have an IP
-          ready? && !public_ip_address.nil? && public_ip_address != '0.0.0.0'
+          print '.'  # Euca instances often report ready before they have an IP
+          is_nil = !public_ip_address.nil? || !private_ip_address.nil?
+          is_zeroes = public_ip_address != '0.0.0.0' || private_ip_address != '0.0.0.0'
```
Comment (tyler-ball) [inline lib/kitchen/driver/ec2.rb]: These variables should be named `not_nil` and `not_zeroes`

**Comment by dtoubelis** [comment]
Sure :-)

**Comment by tyler-ball** [comment]
I already merged another PR that added support for DNS, public and private IP addresses.  

I'm going to merge the ec2 ohai hint code into master, but I'll do it in another PR.  We have some helper methods for the `sudo` logic that we can leverage.

When we merge the Windows/Transport branch to master we will have to update this command to leverage that new feature.

**Comment by tyler-ball** [comment]
Closing - The ohai hints are in https://github.com/test-kitchen/kitchen-ec2/issues/102 and the dns/ip address support was added in https://github.com/test-kitchen/kitchen-ec2/pull/97

**Comment by rmoreira** [comment]
Thank you!

**Pull request closed without merging**